src/source/PeerConnection/SessionDescription.c

```diff
@@ -316,6 +327,74 @@ PCHAR fmtpForPayloadType(UINT64 payloadType, PSessionDescription pSessionDescrip
     return NULL;
 }
 
+/*
+ * Extracts a (hex) value after the provided prefix string. Returns true if
+ * successful.
+ */
+BOOL readHexValue(PCHAR input, PCHAR prefix, PINT32 value)
+{
+    PCHAR substr = STRSTR(input, prefix);
+    if (substr != NULL) {
+        // TODO: This should be changed to SSCANF once the PIC library macro
+        // change is propagated and available here.
+        if (sscanf(substr + STRLEN(prefix), "%x", value) == 1) {
+            return TRUE;
+        }
+    }
+    return FALSE;
+}
+
+/*
+ * Scores the provided fmtp string based on this library's ability to
+ * process various types of H264 streams. A score of 0 indicates an
+ * incompatible fmtp line. Beyond this, a higher score indicates more
+ * compatibility with the desired characteristics, packetization-mode=1,
+ * level-asymmetry-allowed=1, and inbound match with our preferred
+ * profile-level-id. Scoring is on a scale from [0,1], with 1 deemed as
+ * a "perfect fit".
+ *
+ * At some future time, it may be worth expressing this as a true distance
+ * function as defined here, although dealing with infinite floating point
+ * values can get tricky:
+ * https://www.w3.org/TR/mediacapture-streams/#dfn-fitness-distance
+ */
+UINT32 getH264FmtpScore(PCHAR fmtp)
+{
+    INT32 profileId = 0, packetizationMode = 0, levelAsymmetry = 0;
+    BOOL isAsymmetryAllowed = FALSE;
+    BOOL isProfileMatch = FALSE;
+    UINT32 score = 0;
+
+    // No ftmp match found.
+    if (fmtp == NULL) {
+        return 0;
+    }
+
+    // Currently, the packetization mode must be 1, as the packetization logic
+    // is currently not configurable, and sends both NALU and FU-A packets.
+    // https://tools.ietf.org/html/rfc7742#section-6.2
+    if (!readHexValue(fmtp, "packetization-mode=", &packetizationMode) || packetizationMode != 1) {
+        return 0;
+    }
+
+    if (readHexValue(fmtp, "profile-level-id=", &profileId)) {
+        isProfileMatch = (profileId == H264_PROFILE_42E01F);
```

Comment:
You don't need any of the booleans. 

if (readHexValue(fmtp, "profile-level-id=", &profileId) && profileId == H264_PROFILE_42E01F) {
    score++;
}

Reply:
We need to store these booleans because we want to return 0 when the profile doesn't match and it requires a symmetric profile.

Reply:
You don't have to return 0. Remember we talked about weighted normalized multi-dimentional discrete vector? In this case, you add a lot more weight to what's important (after normalizing the units). So, say if we have a profile match then we add 100. When we have a level match we add just 1. At the end of the day, the caller can make a choice of using the score or not given it's absolute vector value.

Reply:
Oh right, that makes sense. I didn't realize that using more weight on the profile match can be used for this case. I'll update it

Reply:
I found that I don't need to add weight to profile or asymmetry dimensions as they're as important as packetization mode.